ledger_device_sdk/src/nbgl.rs

```diff
@@ -99,70 +106,20 @@ pub extern "C" fn io_recv_and_process_event() -> bool {
     false
 }
 
-/// Helper struct that converts strings to null-terminated c strings.
-/// It uses an internal buffer to store the strings, with a maximum size of SIZE.
-struct CStringHelper<const SIZE: usize = 64> {
-    /// Internal buffer where strings are allocated.
-    /// Stored in a [RefCell] because we want [CStringHelper::to_cstring] to be non-mutable.
-    pub buffer: RefCell<[u8; SIZE]>,
-    /// Index of the next string in the internal buffer.
-    /// Stored in a [RefCell] because we want [CStringHelper::to_cstring] to be non-mutable.
-    next: RefCell<usize>,
-}
-
-impl<const SIZE: usize> CStringHelper<SIZE> {
-    pub fn new() -> Self {
-        Self {
-            buffer: RefCell::new([0u8; SIZE]),
-            next: RefCell::new(0),
-        }
-    }
-
-    pub fn flush(&self) {
-        *self.next.borrow_mut() = 0; // Reset the next index to 0
-    }
-
-    pub fn to_cstring<'a>(&'a self, s: &str) -> Result<&'a CStr, ()> {
-        let size = s.len();
-        let mut buffer = self.buffer.borrow_mut();
-        let next: usize = *self.next.borrow();
-        // Verify there is enough space in the internal buffer.
-        // +1 for the null byte
-        if size + next + 1 > buffer.len() {
-            // Not enough space remaining in the internal buffer.
-            return Err(());
-        }
-        // Verify that the input string does not have null bytes already.
-        if s.bytes().find(|c| *c == 0).is_some() {
-            return Err(());
-        }
-
-        // Copy the input string to the internal buffer, and add null byte.
-        buffer[next..next + size].copy_from_slice(s.as_bytes());
-        buffer[next + size] = 0;
-        let start = next;
-        *self.next.borrow_mut() += size + 1;
-
-        let buffer = self.buffer.as_ptr();
-        let slice = unsafe { &(*buffer)[start..start + size + 1] };
-        let cstr = unsafe { CStr::from_bytes_with_nul_unchecked(slice) };
-        Ok(cstr)
-    }
-}
-
 /// Callback triggered by the NBGL API when a setting switch is toggled.
 unsafe fn settings_callback(token: ::core::ffi::c_int, _index: u8, _page: ::core::ffi::c_int) {
     let idx = token - FIRST_USER_TOKEN as i32;
     if idx < 0 || idx >= SETTINGS_SIZE as i32 {
         panic!("Invalid token.");
     }
 
+    let setting_idx: usize = idx as usize;
+    SWITCH_ARRAY[setting_idx].initState = !SWITCH_ARRAY[setting_idx].initState;
+
     if let Some(data) = NVM_REF.as_mut() {
-        let setting_idx: usize = idx as usize;
         let mut switch_values: [u8; SETTINGS_SIZE] = data.get_ref().clone();
         switch_values[setting_idx] = !switch_values[setting_idx];
         data.update(&switch_values);
-        SWITCH_ARRAY[setting_idx].initState = switch_values[setting_idx] as nbgl_state_t;
```

Comment:
I'm not sure we want to separate the true value in the NVM from the SWITCH_ARRAY initState.

Reply:
Initial goal was to separate the settings set in RAM from the settings saved in NVRAM:

1. One can think about an app with settings but no persistent settings: after the app is restarted, the setting are reset. It does not prevent usage of NVRAM to save settings but we could also manage this use case: app with noi persistent settings.
2. Writing systematically in the flash every time a setting is modified can involve an flash wearing. That's why I think we should properly separate the saving of the settings.

Reply:
Ok this has been reverted: when settings are defined, they are systematically saved as NVM data.